runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -362,7 +337,7 @@ end
 --- implementation so it's simply marked @private rather than @deprecated.
 ---
 ---@param client_id integer
----@param buffer_client_map table map of buffers to active clients
+---@param buffer_client_map table<integer, table<integer, table>> map of buffers to active clients
```

Comment:
Can this type be explained?

Reply:
Based on the code it looks like this is a map from buffer to a table of client ID to LSP client.